### DISCUSSION TITLE
Memory usage when importing the library

### DISCUSSION THREAD

**Discussion by flaviobattimo** [moononournation/Arduino_GFX#216] [Q&A]
Hello,
I have a question regarding memory usage when including Arduino_GFX_Library.h

I see many displays are using arrays with a list of commands. These are defined as 
static const uint8_t .....[] = { ...... }

Are these arrays contributing to the RAM usage, or if the display is not used the array is not allocated?

Ask these because I see more RAM usage than what I expected. On an ESP32-S2 I'm not able to use the Canvas (240x240 display) and the board is not able to allocate 115kbytes for the canvas.

Is there a way to only include the required files: GFX, bus, display (st7789) and the canvas? Thank you

**Comment by moononournation** [comment]
You can test empty those array to check memory usage, I expect no memory change if you are not using those class. ESP32 current version seems not allow allocate this big size of memory if no PSRAM exists.

**Reply by flaviobattimo** [reply]
Ok thanks. I think the problem is there is not enough contiguous free memory to allocate 115kbytes. The PSRAM is a solution (need to check if it's available in my ESP32-S2 board). The other way, but more complicated, could be to split the canvas in chunks.
